// @affinidi/affinidi-react-auth/dist/hooks/useInitiateRequest.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const randomstring_1 = __importDefault(require("randomstring"));
const useExtension_1 = require("./useExtension");
function useInitiateRequest({ presentationDefinition, callbackUrl }) {
    const { isInitializing, isExtensionInstalled, client } = (0, useExtension_1.useExtension)();
    async function handleInitiate() {
        if (!isExtensionInstalled) {
            window.location.href = client.getChromeWebStoreUrl();
            return;
        }
        client.initiateAuth({
            nonce: randomstring_1.default.generate(),
            state: randomstring_1.default.generate(),
            responseDestination: {
                responseMode: 'query',
                redirectUri: `${callbackUrl}`,
            },
            presentationDefinition,
        });
    }
    return {
        isInitializing,
        isExtensionInstalled,
        handleInitiate,
    };
}
exports.default = useInitiateRequest;
