// @affinidi/affinidi-react-auth/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useInitiateEmailRequest = exports.useInitiateProfileRequest = exports.useInitiateRequest = exports.useCompleteRequest = exports.useAffinidiProfile = exports.useAffinidiLogin = exports.AffinidiLoginButton = void 0;
const LoginButton_1 = __importDefault(require("./LoginButton"));
exports.AffinidiLoginButton = LoginButton_1.default;
const useAffinidiLogin_1 = __importDefault(require("./hooks/useAffinidiLogin"));
exports.useAffinidiLogin = useAffinidiLogin_1.default;
const useAffinidiProfile_1 = __importDefault(require("./hooks/useAffinidiProfile"));
exports.useAffinidiProfile = useAffinidiProfile_1.default;
const useCompleteRequest_1 = __importDefault(require("./hooks/useCompleteRequest"));
exports.useCompleteRequest = useCompleteRequest_1.default;
const useInitiateRequest_1 = __importDefault(require("./hooks/useInitiateRequest"));
exports.useInitiateRequest = useInitiateRequest_1.default;
const useInitiateProfileRequest_1 = __importDefault(require("./hooks/useInitiateProfileRequest"));
exports.useInitiateProfileRequest = useInitiateProfileRequest_1.default;
const useInitiateEmailRequest_1 = __importDefault(require("./hooks/useInitiateEmailRequest"));
exports.useInitiateEmailRequest = useInitiateEmailRequest_1.default;
__exportStar(require("./types"), exports);
